### PR TITLE
dev: adding counters to bookkeeping

### PR DESCRIPTION
**Pull request by lietava**
This development is related to
https://github.com/AliceO2Group/Bookkeeping/blob/mboulais/O2B-1283/create-cxx-client-for-trigger-counters/cxx-client/README.md
The first problem I can not solve is how to link bookkeeping-api::BookkeepingApi to my O2 code

### CODE REVIEW

**Comment by lietava** [comment]
@ktf , please, can you have a look. I do not know how to link bookeeping library to my O2 code.
I am already discussing with @shahor02 and @martinboulais, but we are stuck.

**Comment by Barthelemy** [comment]
Hi, 

You can see in QC how to link against the BK api. 

1. In your CMakeLists.txt you should find the package : `find_package(BookkeepingApi REQUIRED) # it must be before O2 for some reasons.`
2. In your CMakeLists.txt you should point to the library name when building the binary : `AliceO2::BookkeepingApi` and not `bookkeeping-api::BookkeepingApi`
3. include in your code 
```
#include "BookkeepingApi/BkpClientFactory.h"
#include "BookkeepingApi/BkpClient.h"
```
4. `mClient = BkpClientFactory::create(url);`
5. use `mClient`

**Comment by lietava** [comment]
@Barthelemy : thanks. I can compile now.

**Comment by lietava** [comment]
@Barthelemy , @ktf : how can I "you need to add a dependency on bookkeeping-api in alidist" ?
I added 
 bookkeeping-api
to alidist/o2.sh 
and it seems it is accepted.
Next question is how can I compile my own (martin's) version with API I need to test ?

**Comment by ktf** [comment]
@lietava just clone bookkeeping-api at the same level as O2 and alidist, it will be picked up automatically from your local checkout, rather than the one specified in the recipe. It's a feature which we call "development packages" and which is described at https://alisw.github.io/alibuild/user.html#developing-packages-locally

**Comment by lietava** [comment]
@ktf : thank you for previous response , it worked.

**Comment by lietava** [comment]
@ktf @shahor02 : I can compile locally but at check it says:
CMake Error at DataFormats/Detectors/CTP/CMakeLists.txt:11 (find_package):
  By not providing "FindBookkeepingApi.cmake" in CMAKE_MODULE_PATH this
  project has asked CMake to find a package configuration file provided by
  "BookkeepingApi", but CMake did not find one.
  PLease, let me know how to solve.

**Comment by lietava** [comment]
adding also @Barthelemy

**Comment by lietava** [comment]
@martinboulais suggest to update https://github.com/alisw/alidist/blob/master/o2.sh
?

**Comment by lietava** [comment]
HI @ktf , please, let me know how to proceed here. As I understand the minor fix in alidist/o2.sh is needed or let me know if something to be done on my side.

**Comment by ktf** [comment]
https://github.com/alisw/alidist/pull/5536 should do the job.

**Comment by ktf** [comment]
As a side note, I am not sure having a DataFormat depending on the Bookkeeping API is a good idea. Can't you factorise it into a separate component?

**Comment by lietava** [comment]
@ktf thanks ! 
I can move dependency to Detectors/CTP - would it be better ?

**Comment by ktf** [comment]
Yeah, I would personally do that...

**Comment by lietava** [comment]
@ktf I moved dependency to Detectors/CTP. Now I think there will be conflict with QC which I shall solve.
I still make it to compile to see here.

**Comment by lietava** [comment]
@ktf : I created QC PR to remove RunManager dependence there:
https://github.com/AliceO2Group/QualityControl/pull/2367
If I understand correctly the following sequence of merges is to be done.
1. QC 2367,
2. alidist 5536,
3. O2 13224.
Do I understand correctly ?

**Comment by ktf** [comment]
Your sequence will work. Notice there is no dependency between 1. and 2, so you can do either.

**Comment by lietava** [comment]
@ktf Thanks. Well, then please approve alidist, @Barthelemy or @knopers8 , please, approve QC. Anyway I wait whoever is to approve PRs.

**Comment by lietava** [comment]
Hi @Barthelemy , @shahor02 and @ktf  compilation failed . O2/o2 O2-o2-cs8 failed on macros which I can fix.
But mac failed in cmake  
CMake Error at /opt/homebrew/opt/llvm@15/lib/cmake/llvm/FindgRPC.cmake:85 (find_library):
  Could not find GRPC_LIBRARY using the following names: grpc++ ?
  Any advise on that, please ?

**Comment by Barthelemy** [comment]
@lietava I think that I know what this issue is. Checking

**Comment by Barthelemy** [comment]
@lietava unfortunately I was wrong. I don't know what the issue is.

**Comment by lietava** [comment]
@ktf @shahor02 : now it is clean only mac error.

**Comment by lietava** [comment]
@ktf can you, please, comment on this - it compiles on linuxes but failing for mac.

**Comment by ktf** [comment]
it looks like it's picking up grpc from the system rather than our own.

**Comment by lietava** [comment]
@ktf , thanks.  So you or somebody will fix it ? (It is not superurgent but ~ week time would be  nice).

**Comment by ktf** [comment]
I can have a look in a bit.

**Comment by lietava** [comment]
HI @ktf , anynews here, please ?

**Comment by ktf** [comment]
@lietava I finally managed to give a look. I think you need to have the find_package in the correct place (as done in https://github.com/AliceO2Group/AliceO2/pull/13314) and use CONFIG explicitly. Adding such PR and removing the find_package from yours seems to work for me.

**Comment by lietava** [comment]
@ktf thanks. 
So I should remove find_packege from my CMakeList.txt, then wait for merge if PR 13314 and then try to recompile ?

**Comment by ktf** [comment]
That's what works for me.

**Comment by lietava** [comment]
ok, I changed O2dependencies.cmake and my file locally and it compiles. So waiting for https://github.com/AliceO2Group/AliceO2/pull/13314 merge.

**Comment by lietava** [comment]
HI @ktf , it compiles. 
Now there is problem with macros which do not see new class I factorised out. I have had similar problem in
and solve it by adding new LinkDef.h file:
https://github.com/AliceO2Group/AliceO2/pull/13224/files#diff-51213fb707ff80dd7828b85a5704fdcae681420a05a9d8ef69cd3cdf257667ed
Why does it not work for mac ? Can you, please, have a look, please ?

**Comment by ktf** [comment]
I am on it. I suspect this is an error with the CMakeLists.txt of gRPC...:-(

File: Detectors/CTP/workflowScalers/CMakeLists.txt [hunk 1]
```diff
@@ -8,12 +8,20 @@
 # In applying this license CERN does not waive the privileges and immunities
 # granted to it by virtue of its status as an Intergovernmental Organization
 # or submit itself to any jurisdiction.
+o2_add_library(CTPWorkflowScalers
```
Comment (ktf) [inline Detectors/CTP/workflowScalers/CMakeLists.txt:12]:
```suggestion
if (NOT APPLE)
o2_add_library(CTPWorkflowScalers
```

File: Detectors/CTP/workflowScalers/CMakeLists.txt [hunk 2]
```diff
@@ -8,12 +8,20 @@
...
+                                    AliceO2::BookkeepingApi)
+o2_target_root_dictionary(CTPWorkflowScalers HEADERS
+  include/CTPWorkflowScalers/ctpCCDBManager.h)
 o2_add_executable(
                   proxy
                   COMPONENT_NAME ctp
                   SOURCES src/ctp-proxy.cxx
                   PUBLIC_LINK_LIBRARIES O2::DCStestWorkflow
-                  O2::DataFormatsCTP)
+                  O2::CTPWorkflowScalers)
```
Comment (ktf) [inline Detectors/CTP/workflowScalers/CMakeLists.txt:25]:
```suggestion
                  O2::CTPWorkflowScalers)
endif()
```

**Comment by ktf** [comment]
Only breaks tests on mac. I cannot reproduce, merging to try to better investigate what is going on.

**Comment by lietava** [comment]
@ktf : thank you.

**Comment by ktf** [comment]
After merging this, I actually managed to reproduce the issue in my setup. https://github.com/alisw/alidist/pull/5561 fixes the issue for good.

**Comment by lietava** [comment]
@ktf so I should update alidist pull o2 and recompile ?

**Comment by ktf** [comment]
Only if you are on a mac.

**Pull request merged**